src/composables/useSearchParams.ts

```diff
@@ -62,8 +133,34 @@ export function useSearchParams() {
 
   watchEffect(() => {
     const newFacet = facets.value[facets.value.length - 1];
-    if (newFacet?.field && newFacet?.value) {
-      addFacet();
+    if (newFacet?.field) {
+      if (newFacet.value) {
+        addFacet();
+      } else {
+        if (supportRangeOption(newFacet?.field)) {
+          // assign range values
+          if (!newFacet.range) {
+            newFacet.range = {
+              type:undefined,
+              start: undefined,
+              end: undefined
+            };
+          }
+          switch (newFacet.range.type) {
+          case DateRange.THIS_WEEK:
+          case DateRange.LAST_WEEK:
+          case DateRange.THIS_MONTH:
+          case DateRange.LAST_MONTH:
+            addFacet();
+            break;
+          case DateRange.CUSTOM:
+            if (newFacet.range.start && newFacet.range.end) {
+              addFacet();
+            }
+            break;
+          }
```

Comment:
isn't this the same as 
```suggestion
            addFacet();
```
?